src/main/kotlin/org/jitsi/rtp/ByteArrayBuffer.java

```diff
@@ -89,6 +89,27 @@ public void readRegionToBuff(int off, int len, byte[] outBuff)
         System.arraycopy(this.buffer, startOffset, outBuff, 0, len);
     }
 
+    /**
+     * Append a byte array to the end of the packet. This may change the data
+     * buffer of this packet.
+     *
+     * @param data byte array to append
+     * @param len the number of bytes to append
+     */
+    public void append(byte[] data, int len) {
+        if (data == null || len == 0)  {
+            return;
+        }
+
+        // Ensure the internal buffer is long enough to accommodate data. (The
+        // method grow will re-allocate the internal buffer if it's too short.)
+        grow(len);
```

Comment:
I wonder if it is worth making grow smarter (allowing it to shift the payload left if necessary). It would solve the problem of SRTP re-allocating, but not necessarily in the best way (since we could avoid moving the payload at all if we have space both ways)

Reply:
yeah i think this is worth looking at when i come back around and make the moving of the packet smarter.